src/System.CommandLine/Symbol.cs

```diff
@@ -145,17 +122,38 @@ public bool HasAlias(string alias)
                     .SelectMany(a => a.GetSuggestions(parseResult, textToMatch))
                     .ToArray();
 
-            return this.ChildSymbolAliases()
-                       .Concat(argumentSuggestions)
-                       .Distinct()
-                       .Containing(textToMatch)
-                       .Where(symbol => symbol != null)
-                       .OrderBy(symbol => symbol!.IndexOfCaseInsensitive(textToMatch))
-                       .ThenBy(symbol => symbol, StringComparer.OrdinalIgnoreCase);
+            return Children
+                   .Where(s => !s.IsHidden)
+                   .SelectMany(s => s.RawAliases)
+                   .Concat(argumentSuggestions)
+                   .Distinct()
+                   .Containing(textToMatch)
+                   .Where(symbol => symbol != null)
+                   .OrderBy(symbol => symbol!.IndexOfCaseInsensitive(textToMatch))
+                   .ThenBy(symbol => symbol, StringComparer.OrdinalIgnoreCase);
         }
 
         public override string ToString() => $"{GetType().Name}: {Name}";
 
         ISymbolSet ISymbol.Children => Children;
+
+        internal Action<ISymbol>? OnNameOrAliasChanged;
+
+        [DebuggerStepThrough]
+        private protected void ThrowIfAliasIsInvalid(string alias)
+        {
+            if (string.IsNullOrWhiteSpace(alias))
+            {
+                throw new ArgumentException("An alias cannot be null, empty, or consist entirely of whitespace.");
+            }
+
+            for (var i = 0; i < alias!.Length; i++)
```

Comment:
Is the null forgiveness operator needed here? The parameter is declared non-nullable.